internal/dataplane/dataplane_config.go

```diff
@@ -45,10 +45,11 @@ func (i *GetDataplaneConfigJSONInput) GetDataplaneConfigJSON() ([]byte, error) {
 		},
 	}
 
-	if i.ConsulServerConfig.EnableTLS {
+	grpcTLSSettings := i.ConsulServerConfig.GetGRPCTLSSettings()
+	if grpcTLSSettings.Enabled {
 		cfg.Consul.TLS = &TLSConfig{
-			GRPCCACertPath: i.ConsulServerConfig.CACertFile,
-			TLSServerName:  i.ConsulServerConfig.TLSServerName,
+			GRPCCACertPath: grpcTLSSettings.CaCertFile,
```

Comment:
@pglass It just occurred to me that dataplane always accepts a file path for the CA cert and does not rely directly on a `CA_CERT_PEM` env variable. On the other hand, control plane configures the TLS settings with either a file or a env variable. Couple of options to proceed here

1. Remove ability to pass the cert PEM as a environment variable and only accept a ca cert file path in ECS_CONFIG_JSON.
2. Control plane writes the PEM to a shared volume (if a file input is not provided) and passes the path it to the dataplane as part of the dataplane config json.

I prefer option (2). Let me know your thoughts.

Reply:
I agree. 

I don't think we should remove support for the environment variable because that is how values are securely passed in from Secrets Manager (or Parameter Store[1]) to ECS tasks. There is not a similar native way in AWS to pass values into via _files_ from Secrets Manager, so we need to keep the environment variables for now. (Although the CA cert in question here is technically public / non-sensitive)

https://github.com/hashicorp/terraform-aws-consul-ecs/blob/7440b1a9b1e11769d44232b0475024dccb9bcbd8/modules/mesh-task/main.tf#L259-L270

https://github.com/hashicorp/terraform-aws-consul-ecs/blob/7440b1a9b1e11769d44232b0475024dccb9bcbd8/modules/acl-controller/main.tf#L49-L53

Writing the CA cert to a shared value makes sense, and we can do the same for any other files that are passed to Consul Dataplane (such as the optional client cert/key and prometheus cert files)

---

1. Consul ECS doesn't support Parameter Store right now, but it could and should support it.

Reply:
I can take this up in a separate PR